vulnerabilities/importers/redhat.py

```diff
@@ -35,71 +35,89 @@ class RedhatDataSource(DataSource):
 
 
 def fetch():
-
-    response = []
+    """
+    Return a list of CVE data mappings fetched from the RedHat API.
+    See:
+        https://access.redhat.com/documentation/en-us/red_hat_security_data_api/1.0/html/red_hat_security_data_api/index
+    """
+    cves = []
     page_no = 1
-    url = "https://access.redhat.com/hydra/rest/securitydata/cve.json?page={}"
+    url_template = "https://access.redhat.com/hydra/rest/securitydata/cve.json?per_page=10000&page={}"
 
+    cve_data = None
     while True:
-
-        resp_json = requests.get(url.format(page_no)).json()
-        page_no += 1
-        if not resp_json:
+        current_url = url_template.format(page_no)
+        try:
+            print(f'Fetching: {current_url}')
+            response = requests.get(current_url)
+            if response.status_code != requests.codes.ok:
+                # TODO: log me
+                print(f'Failed to fetch results from {current_url}')
+                break
+            cve_data = response.json()
+        except Exception as e:
+            # TODO: log me
+            msg = f'Failed to fetch results from {current_url}:\n{e}'
+            print(msg)
             break
 
-        for advisory in resp_json:
-            response.append(advisory)
+        if not cve_data:
+            break
+        cves.extend(cve_data)
+        page_no += 1
 
-    return response
+    print(f'Fetched {len(cves)} CVEs from: {current_url}')
+    return cves
 
 
 def to_advisory(advisory_data):
 
     affected_purls = []
     if advisory_data.get("affected_packages"):
         for rpm in advisory_data["affected_packages"]:
-            if rpm_to_purl(rpm):
-                affected_purls.append(rpm_to_purl(rpm))
+            purl = rpm_to_purl(rpm)
+            if purl:
+                affected_purls.append(purl)
 
     references = []
     if advisory_data.get("bugzilla"):
         bugzilla = advisory_data.get("bugzilla")
         references.append(
             Reference(
-                url="https://bugzilla.redhat.com/show_bug.cgi?id={}".format(bugzilla),
                 reference_id=bugzilla,
+                url=f"https://bugzilla.redhat.com/show_bug.cgi?id={bugzilla}",
             )
         )
 
     for rhsa in advisory_data["advisories"]:
         references.append(
             Reference(
-                url="https://access.redhat.com/errata/{}".format(rhsa), reference_id=rhsa,
+                reference_id=rhsa,
+                url="https://access.redhat.com/errata/{rhsa}",
             )
         )
 
     references.append(Reference(url=advisory_data["resource_url"]))
 
     return Advisory(
-        summary=advisory_data["bugzilla_description"],
         cve_id=advisory_data["CVE"],
+        summary=advisory_data["bugzilla_description"],
         impacted_package_urls=affected_purls,
         vuln_references=references,
     )
 
 
 def rpm_to_purl(rpm_string):
+    # FIXME: there is code in scancode to handle RPM conversion AND this should
```

Comment:
I tried nevra.py, but it's horrible here since, redhat has weird names with multiple `-`.  

The method used here is more conservative.

Reply:
> I tried nevra.py, but it's horrible here since, redhat has weird names with multiple -.

I need to see actual real examples of real RPM package names that nevra.py cannot parse. If so this is a serious bug. 

And https://github.com/nexB/vulnerablecode/blob/8efdbd190d1249e5f4bcda044f98ea50af8a86a4/vulnerabilities/tests/test_redhat_importer.py#L43 is not enough tests and foo is a not a real name:  s=you should avoid to make use fake names here IMHO

We really want to have that correct and correctly done in one place only, that we can package in a library if needed.